xds/src/main/java/io/grpc/xds/internal/sds/SdsProtocolNegotiators.java

```diff
@@ -251,23 +254,47 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
     }
   }
 
-  private static final class ServerSdsProtocolNegotiator implements ProtocolNegotiator {
+  @VisibleForTesting
+  public static final class ServerSdsProtocolNegotiator implements ProtocolNegotiator {
 
-    private DownstreamTlsContext downstreamTlsContext;
+    private final DownstreamTlsContext downstreamTlsContext;
     private final XdsClientWrapperForServerSds xdsClientWrapperForServerSds;
 
-    ServerSdsProtocolNegotiator(
-        DownstreamTlsContext downstreamTlsContext, int port, SynchronizationContext syncContext) {
-      this.downstreamTlsContext = downstreamTlsContext;
-      XdsClientWrapperForServerSds localXdsClientWrapperForServerSds;
+    static ProtocolNegotiator createServerNegotiator(
+        @Nullable DownstreamTlsContext downstreamTlsContext,
+        int port,
+        SynchronizationContext syncContext) {
+      XdsClientWrapperForServerSds xdsClientWrapperForServerSds =
+          getXdsClientWrapperForServerSds(port, syncContext);
+      return (xdsClientWrapperForServerSds == null && downstreamTlsContext == null)
+          ? InternalProtocolNegotiators.serverPlaintext()
+          : new ServerSdsProtocolNegotiator(downstreamTlsContext, xdsClientWrapperForServerSds);
+    }
+
+    private static XdsClientWrapperForServerSds getXdsClientWrapperForServerSds(
+        int port, SynchronizationContext syncContext) {
       try {
-        localXdsClientWrapperForServerSds =
-            XdsClientWrapperForServerSds.newInstance(port, Bootstrapper.getInstance(), syncContext);
+        return XdsClientWrapperForServerSds.newInstance(
+            port, Bootstrapper.getInstance(), syncContext);
       } catch (Exception e) {
```

Comment:
(i know we discussed before) this exception seems too aggressive. a bit worried that this can make it very hard to debug